packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -0,0 +1,110 @@
+import { BaseDependencies } from '../config/config.js';
+import categoryRepository from '../repository/categoryRepository.js';
+import submittedRepository from '../repository/submittedRepository.js';
+import submittedUtils from '../utils/submittedDataUtils.js';
+import { SubmittedDataResponse, paginationOps } from '../utils/types.js';
+
+const service = (dependencies: BaseDependencies) => {
+	const LOG_MODULE = 'SUBMITTED_DATA_SERVICE';
+	const submittedDataRepo = submittedRepository(dependencies);
+	const { logger } = dependencies;
+	return {
+		getSubmittedDataByCategory: async (
+			categoryId: number,
+			paginationOps: paginationOps,
+		): Promise<{
+			data: SubmittedDataResponse[];
+			metadata: { totalRecords: number; errorMessage?: string };
+		}> => {
+			const { getSubmittedDataByCategoryIdPaginated, getTotalRecordsByCategoryId } = submittedDataRepo;
+
+			const { categoryIdExists } = categoryRepository(dependencies);
+			const { parseSubmittedData } = submittedUtils(dependencies);
+
+			const isValidCategory = await categoryIdExists(categoryId);
+			if (!isValidCategory) {
+				return {
+					data: [],
+					metadata: {
+						totalRecords: 0,
+						errorMessage: 'Invalid Category ID',
+					},
+				};
```

Comment:
these could be abstracted into a error message dynamic generator

Reply:
added a function `fetchDataErrorResponse` to return this error response object with the corresponding error message